src/_P035_IRTX.ino

```diff
@@ -215,17 +215,17 @@ boolean Plugin_035(byte function, struct EventStruct *event, String& string)
                 for (unsigned int i = 0, total = IrRaw.length(), gotRep = 0, rep = 0; i < total;) {
                    char c = IrRaw[i++];
                    if (c == '*') {
-                       if (i+2 >= total || idx + (rep = from_32hex(IrRaw[i++])) * 2 > sizeof(buf)){
+                       if (i+2 >= total || idx + (rep = from_32hex(IrRaw[i++])) * 2 > sizeof(buf[0])*250){
                          delete[] buf;
                          buf = nullptr;
-                         return addErrorTrue("Invalid RAW2 B32 encoding!");
+                         return addErrorTrue();
                        }
                        gotRep = 2;
                    } else {
-                       if ((c == '^' && i+1 >= total) || idx == sizeof(buf)){
+                       if ((c == '^' && i+1 >= total) || idx == sizeof(buf[0])*250){
```

Comment:
`idx` is also increased by 2 later on in the code.
So you should not compare using `==`, but using `>=`

Also later in the code (starting line 238) there is this part:
```c++
                       if (--gotRep == 0) {
                           while (--rep) {
                               buf[idx] = buf[idx-2];
                               buf[idx+1] = buf[idx-1];
                               idx += 2;
                           }
                       }
```
There is no check for `idx`. There should be a check to see if `idx >=2` and also that `(idx + 1) < length of buf`